fastai/vision/learner.py

```diff
@@ -170,17 +170,20 @@ def _cl_int_plot_top_losses(self, k, largest=True, figsize=(12,12), heatmap:bool
     cols = math.ceil(math.sqrt(k))
     rows = math.ceil(k/cols)
     fig,axes = plt.subplots(rows, cols, figsize=figsize)
-    fig.suptitle('prediction/actual/loss/probability', weight='bold', size=14)
     for i,idx in enumerate(tl_idx):
         im,cl = self.data.dl(self.ds_type).dataset[idx]
         cl = int(cl)
-        im.show(ax=axes.flat[i], title=
-            f'{classes[self.pred_class[idx]]}/{classes[cl]} / {self.losses[idx]:.2f} / {self.preds[idx][cl]:.2f}')
+        if show_text:
+            fig.suptitle('Prediction/Actual/Loss/Probability', weight='bold', size=14)
```

Comment:
This should only be executed once before the loop.

Reply:
Won't this raise an exception because idx is referenced before assignment?

Reply:
I was talking of the line `fig.suptitle`. There is no `idx` in this one.